In [1]:
import pandas as pd
import numpy as np

In [2]:
data_ori = pd.read_csv("Anime.csv")

In [3]:
data_m = pd.read_csv("Anime.csv")

# Understanding the Dataset

In [4]:
data_m.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 18495 entries, 0 to 18494
Data columns (total 17 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Rank             18495 non-null  int64  
 1   Name             18495 non-null  object 
 2   Japanese_name    7938 non-null   object 
 3   Type             18495 non-null  object 
 4   Episodes         9501 non-null   float64
 5   Studio           12018 non-null  object 
 6   Release_season   4116 non-null   object 
 7   Tags             18095 non-null  object 
 8   Rating           15364 non-null  float64
 9   Release_year     18112 non-null  float64
 10  End_year         2854 non-null   float64
 11  Description      18491 non-null  object 
 12  Content_Warning  1840 non-null   object 
 13  Related_Mange    7627 non-null   object 
 14  Related_anime    10063 non-null  object 
 15  Voice_actors     15309 non-null  object 
 16  staff            13005 non-null  object 
dtypes: float64(4

In [5]:
data_m.head()

Rank                                               Name  \
0     1  Demon Slayer: Kimetsu no Yaiba - Entertainment...   
1     2                     Fruits Basket the Final Season   
2     3                                    Mo Dao Zu Shi 3   
3     4                   Fullmetal Alchemist: Brotherhood   
4     5                Attack on Titan 3rd Season: Part II   

                                     Japanese_name   Type  Episodes  \
0                    Kimetsu no Yaiba: Yuukaku-hen  TV          NaN   
1                          Fruits Basket the Final  TV         13.0   
2                       The Founder of Diabolism 3  Web        12.0   
3   Hagane no Renkinjutsushi: Full Metal Alchemist  TV         64.0   
4             Shingeki no Kyojin Season 3: Part II  TV         10.0   

              Studio Release_season  \
0           ufotable          Fall    
1  TMS Entertainment         Spring   
2   B.C MAY PICTURES            NaN   
3              Bones         Spring   
4         WIT Studio         Spring   

                                                Tags  Rating  Release_year  \
0  Action, Adventure, Fantasy, Shounen, Demons, H...    4.60        2021.0   
1  Drama, Fantasy, Romance, Shoujo, Animal Transf...    4.60        2021.0   
2  Fantasy, Ancient China, Chinese Animation, Cul...    4.58        2021.0   
3  Action, Adventure, Drama, Fantasy, Mystery, Sh...    4.58        2009.0   
4  Action, Fantasy, Horror, Shounen, Dark Fantasy...    4.57        2019.0   

   End_year                                        Description  \
0       NaN  'Tanjiro and his friends accompany the Hashira...   
1       NaN                  'The final arc of Fruits Basket.'   
2       NaN               'The third season of Mo Dao Zu Shi.'   
3    2010.0  "The foundation of alchemy is based on the law...   
4       NaN  'The battle to retake Wall Maria begins now! W...   

                                     Content_Warning  \
0                                  Explicit Violence   
1  Emotional Abuse,, Mature Themes,, Physical Abu...   
2                                                NaN   
3  Animal Abuse,, Mature Themes,, Violence,, Dome...   
4                    Cannibalism,, Explicit Violence   

                                       Related_Mange  \
0                     Demon Slayer: Kimetsu no Yaiba   
1               Fruits Basket, Fruits Basket Another   
2  Grandmaster of Demonic Cultivation: Mo Dao Zu ...   
3  Fullmetal Alchemist, Fullmetal Alchemist (Ligh...   
4  Attack on Titan, Attack on Titan: End of the W...   

                                       Related_anime  \
0  Demon Slayer: Kimetsu no Yaiba, Demon Slayer: ...   
1  Fruits Basket 1st Season, Fruits Basket 2nd Se...   
2                   Mo Dao Zu Shi 2, Mo Dao Zu Shi Q   
3  Fullmetal Alchemist: Brotherhood Specials, Ful...   
4  Attack on Titan, Attack on Titan 2nd Season, A...   

                                        Voice_actors  \
0  Inosuke Hashibira : Yoshitsugu Matsuoka, Nezuk...   
1  Akito Sohma : Maaya Sakamoto, Kyo Sohma : Yuum...   
2  Lan Wangji, Wei Wuxian, Jiang Cheng, Jin Guang...   
3  Alphonse Elric : Rie Kugimiya, Edward Elric : ...   
4  Armin Arlelt : Marina Inoue, Eren Jaeger : Yuu...   

                                               staff  
0  Koyoharu Gotouge : Original Creator, Haruo Sot...  
1  Natsuki Takaya : Original Creator, Yoshihide I...  
2  Mo Xiang Tong Xiu : Original Creator, Xiong Ke...  
3  Hiromu Arakawa : Original Creator, Yasuhiro Ir...  
4  Hajime Isayama : Original Creator, Tetsurou Ar...

In [6]:
data_m.tail()

Rank                                               Name Japanese_name  \
18490  18491  Qin Shi Mingyue: Canghai Hengliu Xiaomeng Spec...           NaN   
18491  18492                      Yi Tang Juchang: Sanguo Yanyi           NaN   
18492  18493          Fenghuang Ji Xiang Yu Qingming Shanghe Tu           NaN   
18493  18494                      Chengshi Jiyi Wo Men de Jieri           NaN   
18494  18495  Heisei Inu Monogatari Bow: Genshi Inu Monogata...           NaN   

        Type  Episodes                        Studio Release_season  \
18490  Web         2.0  Sparkly Key Animation Studio            NaN   
18491  TV        108.0                           NaN            NaN   
18492  TV         13.0                           NaN            NaN   
18493  TV          NaN                           NaN            NaN   
18494  Movie       NaN              Nippon Animation            NaN   

                                                    Tags  Rating  \
18490  Action, Ancient China, Chinese Animation, Hist...     NaN   
18491                                  Chinese Animation     NaN   
18492  Chinese Animation, Family Friendly, Short Epis...     NaN   
18493  Chinese Animation, Family Friendly, Short Epis...     NaN   
18494                        Comedy, Slice of Life, Dogs     NaN   

       Release_year  End_year  \
18490        2020.0       NaN   
18491        2010.0       NaN   
18492        2020.0       NaN   
18493        2020.0       NaN   
18494        1994.0       NaN   

                                             Description Content_Warning  \
18490  Special episodes of Qin Shi Mingyue: Canghai H...             NaN   
18491                 No synopsis yet - check back soon!             NaN   
18492                 No synopsis yet - check back soon!             NaN   
18493                 No synopsis yet - check back soon!             NaN   
18494                 No synopsis yet - check back soon!             NaN   

      Related_Mange                     Related_anime Voice_actors staff  
18490           NaN  Qin Shi Mingyue: Canghai Hengliu          NaN   NaN  
18491           NaN                               NaN          NaN   NaN  
18492           NaN                               NaN          NaN   NaN  
18493           NaN                               NaN          NaN   NaN  
18494           NaN         Heisei Inu Monogatari Bow          NaN   NaN

In [7]:
data_m.sample(5)

Rank                                 Name  \
9214    9215                          Honey-Honey   
790      791  Rurouni Kenshin: Special Techniques   
927      928                    Seraph of the End   
16206  16207  Mutsu Gokudo: Mitsudango Kousou Hen   
12379  12380                      Miss Monochrome   

                                           Japanese_name   Type  Episodes  \
9214                     Honey Honey no Suteki na Bouken  TV         29.0   
790                                                  NaN  TV Sp       NaN   
927    Alt titles: Owari no Seraph, Seraph of the End...  TV         12.0   
16206                                                NaN  OVA         2.0   
12379                                                NaN  TV         13.0   

                Studio Release_season  \
9214   Kokusai Eigasha            NaN   
790             Gallop            NaN   
927         WIT Studio         Spring   
16206              NaN            NaN   
12379      LIDEN FILMS          Fall    

                                                    Tags  Rating  \
9214     Adventure, Shoujo, Historical, Based on a Manga    3.27   
790    Action, Shounen, Feudal Japan, Samurai, Swordp...    4.03   
927    Action, Drama, Horror, Shounen, Demons, Pandem...    3.99   
16206              Short Episodes, Stop Motion Animation     NaN   
12379  Comedy, Slice of Life, Androids, Idols, Music,...    3.06   

       Release_year  End_year  \
9214         1981.0    1982.0   
790          1997.0       NaN   
927          2015.0       NaN   
16206        2010.0       NaN   
12379        2013.0       NaN   

                                             Description Content_Warning  \
9214                'No synopsis yet - check back soon!'             NaN   
790                 'No synopsis yet - check back soon!'             NaN   
927    'After a catastrophic epidemic kills every adu...        Violence   
16206                 No synopsis yet - check back soon!             NaN   
12379  Miss Monochrome, an idol who lives in a luxuri...             NaN   

                                           Related_Mange  \
9214                     Honey Honey no Suteki na Bouken   
790    Rurouni Kenshin, Rurouni Kenshin: Voyage to th...   
927    Seraph of the End, Seraph of the End: Guren Ic...   
16206                                                NaN   
12379  Miss Monochrome-san, Miss Monochrome: Motto Ch...   

                                           Related_anime  \
9214                                                 NaN   
790    Rurouni Kenshin, Rurouni Kenshin: Review Speci...   
927    Seraph of the End Specials, Seraph of the End:...   
16206                                                NaN   
12379  Miss Monochrome: Manager, Miss Monochrome: Sup...   

                                            Voice_actors  \
9214   Flora : Fuyumi Shiraishi, Honey Honey : Minori...   
790    Kenshin Himura, Chou Sawagejou, Hajime Saitou,...   
927    Mikaela Hyakuya : Kensho Ono, Shinoa Hiragi : ...   
16206                                                NaN   
12379  Monochrome : Yui Horie, Kikuko : Kikuko Inoue,...   

                                                   staff  
9214   Hideko Mizuno : Original Creator, Takeshi Shir...  
790                  Nobuhiro Watsuki : Original Creator  
927    Takaya Kagami : Original Creator, Daisuke Toku...  
16206                                                NaN  
12379  Yui Horie : Original Character Design, Yoshiak...

##### Looking at the observations

In [8]:
data_m["Tags"][2]

'Fantasy, Ancient China, Chinese Animation, Cultivation, Xianxia, Based on a Web Novel'

In [9]:
data_m["Description"][2]

"'The third season of Mo Dao Zu Shi.'"

In [10]:
data_m["Content_Warning"][2]

nan

In [11]:
data_m["Related_Mange"][2]

'Grandmaster of Demonic Cultivation: Mo Dao Zu Shi (Novel), The Master of Diabolism'

In [12]:
data_m["staff"][0]

'Koyoharu Gotouge : Original Creator, Haruo Sotozaki : Director, Akira Matsushima : Character Design, Aimer : Song Performance'

# Filtering the columns for Model 

In [13]:
# Data for ml algo
# Rank
# Name
# Type
# Tags
# Description
# Staff "Get Original Creator"

In [14]:
data_m = data_m[["Rank", "Name", "Type", "Tags", "Description", "staff"]]

In [15]:
# No duplicates
data_m.duplicated().sum()

0

In [16]:
data_m

Rank                                               Name   Type  \
0          1  Demon Slayer: Kimetsu no Yaiba - Entertainment...  TV      
1          2                     Fruits Basket the Final Season  TV      
2          3                                    Mo Dao Zu Shi 3  Web     
3          4                   Fullmetal Alchemist: Brotherhood  TV      
4          5                Attack on Titan 3rd Season: Part II  TV      
...      ...                                                ...    ...   
18490  18491  Qin Shi Mingyue: Canghai Hengliu Xiaomeng Spec...  Web     
18491  18492                      Yi Tang Juchang: Sanguo Yanyi  TV      
18492  18493          Fenghuang Ji Xiang Yu Qingming Shanghe Tu  TV      
18493  18494                      Chengshi Jiyi Wo Men de Jieri  TV      
18494  18495  Heisei Inu Monogatari Bow: Genshi Inu Monogata...  Movie   

                                                    Tags  \
0      Action, Adventure, Fantasy, Shounen, Demons, H...   
1      Drama, Fantasy, Romance, Shoujo, Animal Transf...   
2      Fantasy, Ancient China, Chinese Animation, Cul...   
3      Action, Adventure, Drama, Fantasy, Mystery, Sh...   
4      Action, Fantasy, Horror, Shounen, Dark Fantasy...   
...                                                  ...   
18490  Action, Ancient China, Chinese Animation, Hist...   
18491                                  Chinese Animation   
18492  Chinese Animation, Family Friendly, Short Epis...   
18493  Chinese Animation, Family Friendly, Short Epis...   
18494                        Comedy, Slice of Life, Dogs   

                                             Description  \
0      'Tanjiro and his friends accompany the Hashira...   
1                      'The final arc of Fruits Basket.'   
2                   'The third season of Mo Dao Zu Shi.'   
3      "The foundation of alchemy is based on the law...   
4      'The battle to retake Wall Maria begins now! W...   
...                                                  ...   
18490  Special episodes of Qin Shi Mingyue: Canghai H...   
18491                 No synopsis yet - check back soon!   
18492                 No synopsis yet - check back soon!   
18493                 No synopsis yet - check back soon!   
18494                 No synopsis yet - check back soon!   

                                                   staff  
0      Koyoharu Gotouge : Original Creator, Haruo Sot...  
1      Natsuki Takaya : Original Creator, Yoshihide I...  
2      Mo Xiang Tong Xiu : Original Creator, Xiong Ke...  
3      Hiromu Arakawa : Original Creator, Yasuhiro Ir...  
4      Hajime Isayama : Original Creator, Tetsurou Ar...  
...                                                  ...  
18490                                                NaN  
18491                                                NaN  
18492                                                NaN  
18493                                                NaN  
18494                                                NaN  

[18495 rows x 6 columns]

In [17]:
data_m.isnull().sum()

Rank              0
Name              0
Type              0
Tags            400
Description       4
staff          5490
dtype: int64

In [18]:
data_m["Type"].value_counts()

TV       5446
Movie    3577
Web      2488
OVA      2235
Music    2165
Other     990
DVD S     911
TV Sp     683
Name: Type, dtype: int64

# Cleaning the data for model

In [19]:
data_m[data_m["Type"] == "TV   "]

Rank                                               Name   Type  \
0          1  Demon Slayer: Kimetsu no Yaiba - Entertainment...  TV      
1          2                     Fruits Basket the Final Season  TV      
3          4                   Fullmetal Alchemist: Brotherhood  TV      
4          5                Attack on Titan 3rd Season: Part II  TV      
5          6                                     Jujutsu Kaisen  TV      
...      ...                                                ...    ...   
18481  18482                                    Goblin Slayer 2  TV      
18488  18489        Shachiku-san wa Youjo Yuurei ni Iyasaretai.  TV      
18491  18492                      Yi Tang Juchang: Sanguo Yanyi  TV      
18492  18493          Fenghuang Ji Xiang Yu Qingming Shanghe Tu  TV      
18493  18494                      Chengshi Jiyi Wo Men de Jieri  TV      

                                                    Tags  \
0      Action, Adventure, Fantasy, Shounen, Demons, H...   
1      Drama, Fantasy, Romance, Shoujo, Animal Transf...   
3      Action, Adventure, Drama, Fantasy, Mystery, Sh...   
4      Action, Fantasy, Horror, Shounen, Dark Fantasy...   
5      Action, Horror, Shounen, Curse, Exorcists, Mon...   
...                                                  ...   
18481  Action, Adventure, Fantasy, Seinen, Dark Fanta...   
18488  Comedy, Slice of Life, Ghosts, Iyashikei, Non-...   
18491                                  Chinese Animation   
18492  Chinese Animation, Family Friendly, Short Epis...   
18493  Chinese Animation, Family Friendly, Short Epis...   

                                             Description  \
0      'Tanjiro and his friends accompany the Hashira...   
1                      'The final arc of Fruits Basket.'   
3      "The foundation of alchemy is based on the law...   
4      'The battle to retake Wall Maria begins now! W...   
5      'Although Yuji Itadori looks like your average...   
...                                                  ...   
18481  It's fall, and the village's harvest festival ...   
18488  The story follows the daily life of Fushihara,...   
18491                 No synopsis yet - check back soon!   
18492                 No synopsis yet - check back soon!   
18493                 No synopsis yet - check back soon!   

                                                   staff  
0      Koyoharu Gotouge : Original Creator, Haruo Sot...  
1      Natsuki Takaya : Original Creator, Yoshihide I...  
3      Hiromu Arakawa : Original Creator, Yasuhiro Ir...  
4      Hajime Isayama : Original Creator, Tetsurou Ar...  
5      Gege Akutami : Original Creator, Seong-Hu Park...  
...                                                  ...  
18481                      Kumo Kagyu : Original Creator  
18488                     Imari Arita : Original Creator  
18491                                                NaN  
18492                                                NaN  
18493                                                NaN  

[5446 rows x 6 columns]

In [20]:
# Filtering the type column by removing spaces and putting it back into the dataframe
temp = data_m["Type"].apply(lambda x: [" ".join(x.split()) for i in x])

In [21]:
def type_(input):
    L = []
    for i in input:
        L.append(i[0])
    return L
        

In [22]:
data_m["Type"]= pd.Series(type_(temp))

In [23]:
data_m[data_m["Type"] == "TV"]

Rank                                               Name Type  \
0          1  Demon Slayer: Kimetsu no Yaiba - Entertainment...   TV   
1          2                     Fruits Basket the Final Season   TV   
3          4                   Fullmetal Alchemist: Brotherhood   TV   
4          5                Attack on Titan 3rd Season: Part II   TV   
5          6                                     Jujutsu Kaisen   TV   
...      ...                                                ...  ...   
18481  18482                                    Goblin Slayer 2   TV   
18488  18489        Shachiku-san wa Youjo Yuurei ni Iyasaretai.   TV   
18491  18492                      Yi Tang Juchang: Sanguo Yanyi   TV   
18492  18493          Fenghuang Ji Xiang Yu Qingming Shanghe Tu   TV   
18493  18494                      Chengshi Jiyi Wo Men de Jieri   TV   

                                                    Tags  \
0      Action, Adventure, Fantasy, Shounen, Demons, H...   
1      Drama, Fantasy, Romance, Shoujo, Animal Transf...   
3      Action, Adventure, Drama, Fantasy, Mystery, Sh...   
4      Action, Fantasy, Horror, Shounen, Dark Fantasy...   
5      Action, Horror, Shounen, Curse, Exorcists, Mon...   
...                                                  ...   
18481  Action, Adventure, Fantasy, Seinen, Dark Fanta...   
18488  Comedy, Slice of Life, Ghosts, Iyashikei, Non-...   
18491                                  Chinese Animation   
18492  Chinese Animation, Family Friendly, Short Epis...   
18493  Chinese Animation, Family Friendly, Short Epis...   

                                             Description  \
0      'Tanjiro and his friends accompany the Hashira...   
1                      'The final arc of Fruits Basket.'   
3      "The foundation of alchemy is based on the law...   
4      'The battle to retake Wall Maria begins now! W...   
5      'Although Yuji Itadori looks like your average...   
...                                                  ...   
18481  It's fall, and the village's harvest festival ...   
18488  The story follows the daily life of Fushihara,...   
18491                 No synopsis yet - check back soon!   
18492                 No synopsis yet - check back soon!   
18493                 No synopsis yet - check back soon!   

                                                   staff  
0      Koyoharu Gotouge : Original Creator, Haruo Sot...  
1      Natsuki Takaya : Original Creator, Yoshihide I...  
3      Hiromu Arakawa : Original Creator, Yasuhiro Ir...  
4      Hajime Isayama : Original Creator, Tetsurou Ar...  
5      Gege Akutami : Original Creator, Seong-Hu Park...  
...                                                  ...  
18481                      Kumo Kagyu : Original Creator  
18488                     Imari Arita : Original Creator  
18491                                                NaN  
18492                                                NaN  
18493                                                NaN  

[5446 rows x 6 columns]

In [24]:
data_m["Type"].value_counts()

TV       5446
Movie    3577
Web      2488
OVA      2235
Music    2165
Other     990
DVD S     911
TV Sp     683
Name: Type, dtype: int64

In [25]:
# Basically observations with no tags and useless Description
# We will drop these from data_m
filtering_1 = data_m[ (data_m["Description"] == "'No synopsis yet - check back soon!'") & (data_m["Tags"].isnull())].index
data_m = data_m.drop(filtering_1)

In [26]:
data_m.isnull().sum()

Rank              0
Name              0
Type              0
Tags            267
Description       4
staff          5451
dtype: int64

In [27]:
# Let us drop the observations which are of type Music, OVA , TV Sp. Since they are mostly a filler anime
filtering_2 = data_m[ (data_m["Type"] == "Music") | (data_m["Type"] == "OVA") | (data_m["Type"] == "TV Sp")].index
data_m.drop(filtering_2,inplace = True)

In [28]:
data_m.isnull().sum()

Rank              0
Name              0
Type              0
Tags             31
Description       4
staff          4322
dtype: int64

In [29]:
filtering_3 = data_m[ (data_m["Description"] == "No synopsis yet - check back soon!") & (data_m["Tags"].isnull())].index
data_m.drop(filtering_3,inplace=True)

In [30]:
data_m.isnull().sum()

Rank              0
Name              0
Type              0
Tags              4
Description       4
staff          4301
dtype: int64

In [31]:
data_m[data_m["Tags"].isnull()]

Rank                                               Name   Type Tags  \
7910    7911                         Himitsu no Akko-chan Movie  Movie  NaN   
8325    8326  Himitsu no Akko-chan: Umi da! Obake da!! Natsu...  Movie  NaN   
8341    8342                        Nanaka 6/17: Ojamana Nanaka  DVD S  NaN   
10185  10186                               Tobe! Kujira no Peek  Movie  NaN   

                                             Description  \
7910   "Taisho's father plans to construct an apartme...   
8325   "Akko spends a summer with her friends at her ...   
8341   'Unaired episode included in the DVD release o...   
10185  'The story of an albino whale used in a circus...   

                                                   staff  
7910   Fujio Akatsuka : Original Creator, Hiroki Shib...  
8325   Fujio Akatsuka : Original Creator, Hiroki Shib...  
8341                                                 NaN  
10185  Kouji Morimoto : Director, Satoru Utsunomiya :...

In [32]:
# we know that the tags do play a crucial role in recommendations. So the observations with Rank 7911, 8325, 8342, 10186.
# WILL BE REMOVED
filtering_4 = data_m[data_m["Tags"].isnull()].index
data_m.drop(filtering_4,inplace=True)

In [33]:
data_m.isnull().sum()

Rank              0
Name              0
Type              0
Tags              0
Description       4
staff          4300
dtype: int64

In [34]:
data_m[data_m["Description"] == "No synopsis yet - check back soon!"]

Rank                                               Name   Type  \
10505  10506                  Sekai Meisaku Douwa: Oyayubi Hime  Movie   
10506  10507                Inugami-san to Nekoyama-san Special  DVD S   
10509  10510                              Lupin III: Pilot Film  Other   
10513  10514                                  Umbrella Fighter!    Web   
10515  10516                Inishie no Megami to Houseki no Ite    Web   
...      ...                                                ...    ...   
18479  18480                                     Baise Shandian    Web   
18491  18492                      Yi Tang Juchang: Sanguo Yanyi     TV   
18492  18493          Fenghuang Ji Xiang Yu Qingming Shanghe Tu     TV   
18493  18494                      Chengshi Jiyi Wo Men de Jieri     TV   
18494  18495  Heisei Inu Monogatari Bow: Genshi Inu Monogata...  Movie   

                                                    Tags  \
10505                           Fantasy, Family Friendly   
10506  Comedy, GL, Shoujo-ai, Slice of Life, Based on...   
10509  Action, Comedy, Criminals, Shorts, Thieves, Ba...   
10513  Slice of Life, No Dialogue, Short Episodes, Fl...   
10515  Action, Fantasy, Commercials, Dragons, Promoti...   
...                                                  ...   
18479               Sports, Chinese Animation, Ping Pong   
18491                                  Chinese Animation   
18492  Chinese Animation, Family Friendly, Short Epis...   
18493  Chinese Animation, Family Friendly, Short Epis...   
18494                        Comedy, Slice of Life, Dogs   

                              Description  \
10505  No synopsis yet - check back soon!   
10506  No synopsis yet - check back soon!   
10509  No synopsis yet - check back soon!   
10513  No synopsis yet - check back soon!   
10515  No synopsis yet - check back soon!   
...                                   ...   
18479  No synopsis yet - check back soon!   
18491  No synopsis yet - check back soon!   
18492  No synopsis yet - check back soon!   
18493  No synopsis yet - check back soon!   
18494  No synopsis yet - check back soon!   

                                                   staff  
10505  Shunsuke Kikuchi : Music, Osamu Tezuka : Chara...  
10506                       Kuzushiro : Original Creator  
10509                    Monkey Punch : Original Creator  
10513                                                NaN  
10515                                                NaN  
...                                                  ...  
18479                                                NaN  
18491                                                NaN  
18492                                                NaN  
18493                                                NaN  
18494                                                NaN  

[3235 rows x 6 columns]

In [35]:
data_m[data_m["Description"].isnull()]

Rank                    Name Type  \
10910  10911  Chiisana Viking Vickie   TV   
15401  15402              Jingju Mao  Web   
15473  15474    Pororo Goes to Korea  Web   
16393  16394                Bad Mood  Web   

                                                    Tags Description  \
10910  Adventure, Comedy, Historical, Ocean, Based on...         NaN   
15401  Action, Adventure, Fantasy, Animal Protagonist...         NaN   
15473  Adventure, Animal Protagonists, Birds, Family ...         NaN   
16393  Slice of Life, Music, Non-Human Protagonists, ...         NaN   

                                                   staff  
10910  Hiroshi Saito : Director, Shuichi Seki : Chara...  
15401                                                NaN  
15473                                                NaN  
16393                                                NaN

In [36]:
data_m[data_m["staff"].isnull()]

Rank                                               Name   Type  \
47        48       Demon Slayer: Kimetsu no Yaiba - Mugen Train     TV   
122      123                                  The Legend of Hei  Movie   
162      163                                Hikari: Be My Light    Web   
208      209             Yuri!!! on Ice: Welcome to The Madness  DVD S   
234      235                         School Babysitters Special  DVD S   
...      ...                                                ...    ...   
18490  18491  Qin Shi Mingyue: Canghai Hengliu Xiaomeng Spec...    Web   
18491  18492                      Yi Tang Juchang: Sanguo Yanyi     TV   
18492  18493          Fenghuang Ji Xiang Yu Qingming Shanghe Tu     TV   
18493  18494                      Chengshi Jiyi Wo Men de Jieri     TV   
18494  18495  Heisei Inu Monogatari Bow: Genshi Inu Monogata...  Movie   

                                                    Tags  \
47     Action, Drama, Fantasy, Shounen, Demons, Histo...   
122    Action, Adventure, Fantasy, Animal Characteris...   
162    BL, Drama, Shounen-ai, Illness, Psychological,...   
208                   Sports, Ice Skating, Original Work   
234    Comedy, Shoujo, Slice of Life, Childcare, Life...   
...                                                  ...   
18490  Action, Ancient China, Chinese Animation, Hist...   
18491                                  Chinese Animation   
18492  Chinese Animation, Family Friendly, Short Epis...   
18493  Chinese Animation, Family Friendly, Short Epis...   
18494                        Comedy, Slice of Life, Dogs   

                                             Description staff  
47     'Tanjirou\xa0and the group have completed thei...   NaN  
122    'In bustling human world, various of goblins l...   NaN  
162                 'No synopsis yet - check back soon!'   NaN  
208                 'No synopsis yet - check back soon!'   NaN  
234                 'No synopsis yet - check back soon!'   NaN  
...                                                  ...   ...  
18490  Special episodes of Qin Shi Mingyue: Canghai H...   NaN  
18491                 No synopsis yet - check back soon!   NaN  
18492                 No synopsis yet - check back soon!   NaN  
18493                 No synopsis yet - check back soon!   NaN  
18494                 No synopsis yet - check back soon!   NaN  

[4300 rows x 6 columns]

In [37]:
data_m.count()

Rank           13356
Name           13356
Type           13356
Tags           13356
Description    13352
staff           9056
dtype: int64

In [38]:
data_m.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 13356 entries, 0 to 18494
Data columns (total 6 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   Rank         13356 non-null  int64 
 1   Name         13356 non-null  object
 2   Type         13356 non-null  object
 3   Tags         13356 non-null  object
 4   Description  13352 non-null  object
 5   staff        9056 non-null   object
dtypes: int64(1), object(5)
memory usage: 730.4+ KB


In [39]:
data_m["Description"][47]

"'Tanjirou\\xa0and the group have completed their rehabilitation training at the Butterfly Mansion, and they arrive to their next mission on the Mugen Train, where over 40 people have disappeared in a very short period of time. Tanjirou and\\xa0Nezuko, along with\\xa0Zenitsu\\xa0and\\xa0Inosuke, join one of the most powerful swordsmen within the Demon Slayer Corps,\\xa0Flame Hashira Rengoku Kyoujurou, to face the demon aboard the Mugen Train on track to despair.'"

In [40]:
data_m["Tags"][47]

'Action, Drama, Fantasy, Shounen, Demons, Historical, Orphans, Siblings, Swordplay, Trains, Based on a Manga, Mature Themes,, Suicide,, Violence'

In [41]:
# Upon looking at the Description column. We have noticed that not all the observations have descriptions
# If we look at it most of them are 'No synopsis yet - check back soon!'   OR
# No synopsis yet - check back soon!    as the description.

data_m.drop("Description", axis=1, inplace=True)

In [42]:
data_m.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 13356 entries, 0 to 18494
Data columns (total 5 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   Rank    13356 non-null  int64 
 1   Name    13356 non-null  object
 2   Type    13356 non-null  object
 3   Tags    13356 non-null  object
 4   staff   9056 non-null   object
dtypes: int64(1), object(4)
memory usage: 1.1+ MB


Upon going thorough the staff variable.
- We have noticed that there are 5490 missing values. We could get rid of the column entirely.
- However in the world of Anime. Original Creaters like Eiichiro Oda (One Piece), Hajime Isayama (Attack on Titan), Masashi Kishimoto (Naruto), many more have a great Manga sales throughout the world.

What is so great about having sales for considering Original Creaters name?

There are people who are willing to watch the anime based on the anime Original Creater. Hence, we will use it as one of the features

**Therefore, we will only keeping the observations with staff variable present which containd the Original Creater Name.**


In [43]:
data_m["staff"].isna().sum()

4300

# Filtering of Staff variable

In [44]:
# Filtering out the values that has staff
data_m_mini = data_m[data_m["staff"].notnull()]

In [45]:
data_m_mini.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 9056 entries, 0 to 18489
Data columns (total 5 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   Rank    9056 non-null   int64 
 1   Name    9056 non-null   object
 2   Type    9056 non-null   object
 3   Tags    9056 non-null   object
 4   staff   9056 non-null   object
dtypes: int64(1), object(4)
memory usage: 424.5+ KB


In [46]:
data_m_mini.isnull().sum()

Rank     0
Name     0
Type     0
Tags     0
staff    0
dtype: int64

In [47]:
data_m_mini["staff"].isna().sum()

0

In [48]:
data_m_mini["staff"][0]

'Koyoharu Gotouge : Original Creator, Haruo Sotozaki : Director, Akira Matsushima : Character Design, Aimer : Song Performance'

In [49]:
type(data_m_mini["staff"][0])

str

In [50]:
type(data_m_mini["staff"][0])

str

In [51]:
type(data_m_mini["staff"])

pandas.core.series.Series

In [52]:
# We need to convert these to dict
data_m_mini["staff"][1]

'Natsuki Takaya : Original Creator, Yoshihide Ibata : Director & Episode Director & Storyboard, Taku Kishimoto : Screenplay & Series Composition, Masaru Yokoyama : Music, Masaru Shindou : Character Design & Chief Animation Director, Baek-Ryun Chae : Photography Director, Youko Koyama : Art Director, Mika Sugawara : Color Design'

In [53]:
data_m_mini

Rank                                               Name   Type  \
0          1  Demon Slayer: Kimetsu no Yaiba - Entertainment...     TV   
1          2                     Fruits Basket the Final Season     TV   
2          3                                    Mo Dao Zu Shi 3    Web   
3          4                   Fullmetal Alchemist: Brotherhood     TV   
4          5                Attack on Titan 3rd Season: Part II     TV   
...      ...                                                ...    ...   
18484  18485                                           Tooriame  Movie   
18486  18487                                   Better back then  Movie   
18487  18488                                        Make My Day    Web   
18488  18489        Shachiku-san wa Youjo Yuurei ni Iyasaretai.     TV   
18489  18490                                          Nostalgia  Movie   

                                                    Tags  \
0      Action, Adventure, Fantasy, Shounen, Demons, H...   
1      Drama, Fantasy, Romance, Shoujo, Animal Transf...   
2      Fantasy, Ancient China, Chinese Animation, Cul...   
3      Action, Adventure, Drama, Fantasy, Mystery, Sh...   
4      Action, Fantasy, Horror, Shounen, Dark Fantasy...   
...                                                  ...   
18484                                   Abstract, Shorts   
18486       Shorts, Stop Motion Animation, Original Work   
18487                    Horror, Monsters, Original Work   
18488  Comedy, Slice of Life, Ghosts, Iyashikei, Non-...   
18489  Drama, No Dialogue, Shorts, Stop Motion Animation   

                                                   staff  
0      Koyoharu Gotouge : Original Creator, Haruo Sot...  
1      Natsuki Takaya : Original Creator, Yoshihide I...  
2      Mo Xiang Tong Xiu : Original Creator, Xiong Ke...  
3      Hiromu Arakawa : Original Creator, Yasuhiro Ir...  
4      Hajime Isayama : Original Creator, Tetsurou Ar...  
...                                                  ...  
18484  Tomoko Takayama : Director, Juri Ito : Music, ...  
18486  Naoaki Shibuta : Director, Chikara Uemizutaru ...  
18487  Yasuo Ootagaki : Original Creator, Makoto Kane...  
18488                     Imari Arita : Original Creator  
18489                         Tomoyasu Murata : Director  

[9056 rows x 5 columns]

In [54]:
# Let use reset the index for the staff conversion from a Series str to a Series of dict
data_m_mini.reset_index(inplace = True)

In [55]:
#### SAMPLE #####
# Convert to dict
try_1 = data_m_mini["staff"][1].split(",")
try_1 = {k.split(":")[0].strip() : k.split(":")[1].strip() for k in try_1}
print(try_1)

{'Natsuki Takaya': 'Original Creator', 'Yoshihide Ibata': 'Director & Episode Director & Storyboard', 'Taku Kishimoto': 'Screenplay & Series Composition', 'Masaru Yokoyama': 'Music', 'Masaru Shindou': 'Character Design & Chief Animation Director', 'Baek-Ryun Chae': 'Photography Director', 'Youko Koyama': 'Art Director', 'Mika Sugawara': 'Color Design'}


In [56]:
# a Series of  str to Dict
for i in range (0,len(data_m_mini["staff"])):
    try_1 = data_m_mini["staff"][i].split(",")
    data_m_mini["staff"][i] = {k.split(":")[0].strip() : k.split(":")[1].strip() for k in try_1}

/var/folders/_w/3rp8x21n1fzc11xml_g_c1t80000gn/T/ipykernel_13641/2632981570.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data_m_mini["staff"][i] = {k.split(":")[0].strip() : k.split(":")[1].strip() for k in try_1}


In [57]:
data_m_mini

index   Rank                                               Name   Type  \
0         0      1  Demon Slayer: Kimetsu no Yaiba - Entertainment...     TV   
1         1      2                     Fruits Basket the Final Season     TV   
2         2      3                                    Mo Dao Zu Shi 3    Web   
3         3      4                   Fullmetal Alchemist: Brotherhood     TV   
4         4      5                Attack on Titan 3rd Season: Part II     TV   
...     ...    ...                                                ...    ...   
9051  18484  18485                                           Tooriame  Movie   
9052  18486  18487                                   Better back then  Movie   
9053  18487  18488                                        Make My Day    Web   
9054  18488  18489        Shachiku-san wa Youjo Yuurei ni Iyasaretai.     TV   
9055  18489  18490                                          Nostalgia  Movie   

                                                   Tags  \
0     Action, Adventure, Fantasy, Shounen, Demons, H...   
1     Drama, Fantasy, Romance, Shoujo, Animal Transf...   
2     Fantasy, Ancient China, Chinese Animation, Cul...   
3     Action, Adventure, Drama, Fantasy, Mystery, Sh...   
4     Action, Fantasy, Horror, Shounen, Dark Fantasy...   
...                                                 ...   
9051                                   Abstract, Shorts   
9052       Shorts, Stop Motion Animation, Original Work   
9053                    Horror, Monsters, Original Work   
9054  Comedy, Slice of Life, Ghosts, Iyashikei, Non-...   
9055  Drama, No Dialogue, Shorts, Stop Motion Animation   

                                                  staff  
0     {'Koyoharu Gotouge': 'Original Creator', 'Haru...  
1     {'Natsuki Takaya': 'Original Creator', 'Yoshih...  
2     {'Mo Xiang Tong Xiu': 'Original Creator', 'Xio...  
3     {'Hiromu Arakawa': 'Original Creator', 'Yasuhi...  
4     {'Hajime Isayama': 'Original Creator', 'Tetsur...  
...                                                 ...  
9051  {'Tomoko Takayama': 'Director', 'Juri Ito': 'M...  
9052  {'Naoaki Shibuta': 'Director', 'Chikara Uemizu...  
9053  {'Yasuo Ootagaki': 'Original Creator', 'Makoto...  
9054                {'Imari Arita': 'Original Creator'}  
9055                    {'Tomoyasu Murata': 'Director'}  

[9056 rows x 6 columns]

In [58]:
data_m_mini["staff"].duplicated().sum()

1416

In [59]:
data_m_mini = data_m_mini[["Rank","Name","Tags", "staff"]]

In [60]:
data_m_mini

Rank                                               Name  \
0         1  Demon Slayer: Kimetsu no Yaiba - Entertainment...   
1         2                     Fruits Basket the Final Season   
2         3                                    Mo Dao Zu Shi 3   
3         4                   Fullmetal Alchemist: Brotherhood   
4         5                Attack on Titan 3rd Season: Part II   
...     ...                                                ...   
9051  18485                                           Tooriame   
9052  18487                                   Better back then   
9053  18488                                        Make My Day   
9054  18489        Shachiku-san wa Youjo Yuurei ni Iyasaretai.   
9055  18490                                          Nostalgia   

                                                   Tags  \
0     Action, Adventure, Fantasy, Shounen, Demons, H...   
1     Drama, Fantasy, Romance, Shoujo, Animal Transf...   
2     Fantasy, Ancient China, Chinese Animation, Cul...   
3     Action, Adventure, Drama, Fantasy, Mystery, Sh...   
4     Action, Fantasy, Horror, Shounen, Dark Fantasy...   
...                                                 ...   
9051                                   Abstract, Shorts   
9052       Shorts, Stop Motion Animation, Original Work   
9053                    Horror, Monsters, Original Work   
9054  Comedy, Slice of Life, Ghosts, Iyashikei, Non-...   
9055  Drama, No Dialogue, Shorts, Stop Motion Animation   

                                                  staff  
0     {'Koyoharu Gotouge': 'Original Creator', 'Haru...  
1     {'Natsuki Takaya': 'Original Creator', 'Yoshih...  
2     {'Mo Xiang Tong Xiu': 'Original Creator', 'Xio...  
3     {'Hiromu Arakawa': 'Original Creator', 'Yasuhi...  
4     {'Hajime Isayama': 'Original Creator', 'Tetsur...  
...                                                 ...  
9051  {'Tomoko Takayama': 'Director', 'Juri Ito': 'M...  
9052  {'Naoaki Shibuta': 'Director', 'Chikara Uemizu...  
9053  {'Yasuo Ootagaki': 'Original Creator', 'Makoto...  
9054                {'Imari Arita': 'Original Creator'}  
9055                    {'Tomoyasu Murata': 'Director'}  

[9056 rows x 4 columns]

In [61]:
data_m_mini["staff"][1].keys()

dict_keys(['Natsuki Takaya', 'Yoshihide Ibata', 'Taku Kishimoto', 'Masaru Yokoyama', 'Masaru Shindou', 'Baek-Ryun Chae', 'Youko Koyama', 'Mika Sugawara'])

In [62]:
data_m_mini["staff"][1].values()

dict_values(['Original Creator', 'Director & Episode Director & Storyboard', 'Screenplay & Series Composition', 'Music', 'Character Design & Chief Animation Director', 'Photography Director', 'Art Director', 'Color Design'])

In [63]:
#### Sample ######
dict = {'a': 1, 'b': 2, 'c': 3}
print(dict)

dict = {value:key for key, value in dict.items()}
print(dict)

{'a': 1, 'b': 2, 'c': 3}
{1: 'a', 2: 'b', 3: 'c'}


In [64]:
# Converting values to keys and keys to values in Dict
for i in range (0,len(data_m_mini["staff"])):
    dict = data_m_mini["staff"][i]
    data_m_mini["staff"][i] = {value:key for key, value in dict.items()}

/var/folders/_w/3rp8x21n1fzc11xml_g_c1t80000gn/T/ipykernel_13641/2305311921.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data_m_mini["staff"][i] = {value:key for key, value in dict.items()}


In [65]:
data_m_mini

Rank                                               Name  \
0         1  Demon Slayer: Kimetsu no Yaiba - Entertainment...   
1         2                     Fruits Basket the Final Season   
2         3                                    Mo Dao Zu Shi 3   
3         4                   Fullmetal Alchemist: Brotherhood   
4         5                Attack on Titan 3rd Season: Part II   
...     ...                                                ...   
9051  18485                                           Tooriame   
9052  18487                                   Better back then   
9053  18488                                        Make My Day   
9054  18489        Shachiku-san wa Youjo Yuurei ni Iyasaretai.   
9055  18490                                          Nostalgia   

                                                   Tags  \
0     Action, Adventure, Fantasy, Shounen, Demons, H...   
1     Drama, Fantasy, Romance, Shoujo, Animal Transf...   
2     Fantasy, Ancient China, Chinese Animation, Cul...   
3     Action, Adventure, Drama, Fantasy, Mystery, Sh...   
4     Action, Fantasy, Horror, Shounen, Dark Fantasy...   
...                                                 ...   
9051                                   Abstract, Shorts   
9052       Shorts, Stop Motion Animation, Original Work   
9053                    Horror, Monsters, Original Work   
9054  Comedy, Slice of Life, Ghosts, Iyashikei, Non-...   
9055  Drama, No Dialogue, Shorts, Stop Motion Animation   

                                                  staff  
0     {'Original Creator': 'Koyoharu Gotouge', 'Dire...  
1     {'Original Creator': 'Natsuki Takaya', 'Direct...  
2     {'Original Creator': 'Mo Xiang Tong Xiu', 'Chi...  
3     {'Original Creator': 'Hiromu Arakawa', 'Direct...  
4     {'Original Creator': 'Hajime Isayama', 'Chief ...  
...                                                 ...  
9051  {'Director': 'Tomoko Takayama', 'Music': 'Juri...  
9052  {'Director': 'Naoaki Shibuta', 'Music': 'Chika...  
9053  {'Original Creator': 'Yasuo Ootagaki', 'Direct...  
9054                {'Original Creator': 'Imari Arita'}  
9055                    {'Director': 'Tomoyasu Murata'}  

[9056 rows x 4 columns]

In [66]:
data_m_mini.iloc[8000:8023,3:4]

staff
8000  {'Director': 'Naoyuki Tsuji', 'Music': 'Makiko...
8001  {'Director': 'Naoyuki Kuzuya', 'Series Composi...
8002                    {'Director': 'Kazuhiro Hocchi'}
8003  {'Director': 'Toshirou Hamamura', 'Music': 'Sh...
8004  {'Director & Character Design & Original Creat...
8005  {'Director & Character Design': 'Takashi Tanig...
8006  {'Original Creator': 'Daisuke Kimura', 'Direct...
8007                     {'Director': 'Keita Kurosaka'}
8008  {'Director': 'Ryuji Masuda', 'Producer': 'Shun...
8009  {'Director': 'Yuuichi Abe', 'Music': 'Kenichir...
8010                  {'Director & Music': 'Tact Aoki'}
8011  {'Original Creator': 'Kouta Sannomiya', 'Origi...
8012  {'Director': 'Sawako Kabuki', 'Music': 'Saeki ...
8013  {'Director': 'Yuutarou Kubo', 'Music': 'Miki S...
8014                          {'Director': 'Yoji Kuri'}
8015                     {'Producer': 'Tomohiro Ogawa'}
8016  {'Director & Screenplay & Series Composition &...
8017  {'Director': 'Hitomi Ohtakara', 'Music': 'Mika...
8018  {'Director': 'Hisashi Ishii', 'Character Desig...
8019  {'Character Design': 'Atsuko Takahashi', 'Prod...
8020  {'Director': 'Souta Sugahara', 'Music': 'Takaf...
8021                        {'Director': 'Tokio Heidi'}
8022                       {'Director': 'Takena Nagao'}

In [67]:
data_m_mini["staff"][1].keys()

dict_keys(['Original Creator', 'Director & Episode Director & Storyboard', 'Screenplay & Series Composition', 'Music', 'Character Design & Chief Animation Director', 'Photography Director', 'Art Director', 'Color Design'])

In [68]:
# Filtering for Original Creators in staff Column
for i in range (0,len(data_m_mini["staff"])):
    data_m_mini["staff"][i] = { key:value for (key,value) in data_m_mini["staff"][i].items() if key  == "Original Creator"}

/var/folders/_w/3rp8x21n1fzc11xml_g_c1t80000gn/T/ipykernel_13641/2570665235.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data_m_mini["staff"][i] = { key:value for (key,value) in data_m_mini["staff"][i].items() if key  == "Original Creator"}


In [69]:
data_m_mini.iloc[1:40,3:4]

staff
1      {'Original Creator': 'Natsuki Takaya'}
2   {'Original Creator': 'Mo Xiang Tong Xiu'}
3      {'Original Creator': 'Hiromu Arakawa'}
4      {'Original Creator': 'Hajime Isayama'}
5        {'Original Creator': 'Gege Akutami'}
6      {'Original Creator': 'Hajime Isayama'}
7      {'Original Creator': 'Hajime Isayama'}
8    {'Original Creator': 'Koyoharu Gotouge'}
9   {'Original Creator': 'Haruichi Furudate'}
10                                         {}
11  {'Original Creator': 'Haruichi Furudate'}
12   {'Original Creator': 'Koyoharu Gotouge'}
13  {'Original Creator': 'Yoshihiro Togashi'}
14    {'Original Creator': 'Yoshitoki Ooima'}
15      {'Original Creator': 'Kana Akatsuki'}
16  {'Original Creator': 'Haruichi Furudate'}
17  {'Original Creator': 'Mo Xiang Tong Xiu'}
18               {'Original Creator': 'Fuse'}
19       {'Original Creator': 'Natsuki Kizu'}
20                                         {}
21  {'Original Creator': 'Mo Xiang Tong Xiu'}
22    {'Original Creator': 'Hideaki Sorachi'}
23                                         {}
24                {'Original Creator': 'One'}
25  {'Original Creator': 'Haruichi Furudate'}
26     {'Original Creator': 'Natsuki Takaya'}
27        {'Original Creator': 'Kaiu Shirai'}
28                                         {}
29               {'Original Creator': 'Fuse'}
30  {'Original Creator': 'Mo Xiang Tong Xiu'}
31       {'Original Creator': 'Sousuke Toka'}
32                                         {}
33    {'Original Creator': 'Hideaki Sorachi'}
34                                         {}
35  {'Original Creator': 'Mo Xiang Tong Xiu'}
36    {'Original Creator': 'Hideaki Sorachi'}
37      {'Original Creator': 'Akimi Yoshida'}
38                                         {}
39                                         {}

In [70]:
# filtering for dict with values
filtering_5 = data_m_mini[data_m_mini["staff"] == {}].index
data_m_mini.drop(filtering_5,inplace = True)

/var/folders/_w/3rp8x21n1fzc11xml_g_c1t80000gn/T/ipykernel_13641/544382922.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data_m_mini.drop(filtering_5,inplace = True)


In [71]:
# Reset index aagin
data_m_mini.reset_index(inplace = True)

In [72]:
# Keeping the values of Original Creator
for i in range (0,len(data_m_mini["staff"])):
        data_m_mini["staff"][i] = (data_m_mini["staff"][i])["Original Creator"]

/var/folders/_w/3rp8x21n1fzc11xml_g_c1t80000gn/T/ipykernel_13641/1092330540.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data_m_mini["staff"][i] = (data_m_mini["staff"][i])["Original Creator"]


In [73]:
############ We have succesfully extracted the Original Creater Name ##########

In [74]:
data_m_mini

index   Rank                                               Name  \
0         0      1  Demon Slayer: Kimetsu no Yaiba - Entertainment...   
1         1      2                     Fruits Basket the Final Season   
2         2      3                                    Mo Dao Zu Shi 3   
3         3      4                   Fullmetal Alchemist: Brotherhood   
4         4      5                Attack on Titan 3rd Season: Part II   
...     ...    ...                                                ...   
3194   9037  18444                                      The Imaginary   
3195   9049  18482                                    Goblin Slayer 2   
3196   9050  18484                        Peleliu: Rakuen no Guernica   
3197   9053  18488                                        Make My Day   
3198   9054  18489        Shachiku-san wa Youjo Yuurei ni Iyasaretai.   

                                                   Tags              staff  
0     Action, Adventure, Fantasy, Shounen, Demons, H...   Koyoharu Gotouge  
1     Drama, Fantasy, Romance, Shoujo, Animal Transf...     Natsuki Takaya  
2     Fantasy, Ancient China, Chinese Animation, Cul...  Mo Xiang Tong Xiu  
3     Action, Adventure, Drama, Fantasy, Mystery, Sh...     Hiromu Arakawa  
4     Action, Fantasy, Horror, Shounen, Dark Fantasy...     Hajime Isayama  
...                                                 ...                ...  
3194                                   Based on a Novel      A. F. Harrold  
3195  Action, Adventure, Fantasy, Seinen, Dark Fanta...         Kumo Kagyu  
3196  Action, Shounen, Island, War, World War 2, Bas...    Masao Hiratsuka  
3197                    Horror, Monsters, Original Work     Yasuo Ootagaki  
3198  Comedy, Slice of Life, Ghosts, Iyashikei, Non-...        Imari Arita  

[3199 rows x 5 columns]

In [75]:
data_m_mini["Tags"] = data_m_mini["Tags"] +", "+ data_m_mini["staff"]

/var/folders/_w/3rp8x21n1fzc11xml_g_c1t80000gn/T/ipykernel_13641/2256362315.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data_m_mini["Tags"] = data_m_mini["Tags"] +", "+ data_m_mini["staff"]


In [76]:
data_m_mini.drop(labels = "staff",axis = 1, inplace= True)

/var/folders/_w/3rp8x21n1fzc11xml_g_c1t80000gn/T/ipykernel_13641/3827376557.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data_m_mini.drop(labels = "staff",axis = 1, inplace= True)


In [77]:
data_m.drop("staff",axis = 1, inplace = True)

In [78]:
data_m_mini["Tags"][0]

'Action, Adventure, Fantasy, Shounen, Demons, Historical, Martial Arts, Orphans, Siblings, Swordplay, Based on a Manga, Explicit Violence, Koyoharu Gotouge'

In [79]:
data_m["Tags"][0]

'Action, Adventure, Fantasy, Shounen, Demons, Historical, Martial Arts, Orphans, Siblings, Swordplay, Based on a Manga, Explicit Violence'

In [80]:
data_m["Tags"][1].replace(data_m["Tags"][1],data_m_mini["Tags"][1])

'Drama, Fantasy, Romance, Shoujo, Animal Transformation, Contemporary Fantasy, Curse, Dysfunctional Families, Mental Illness, Orphans, Based on a Manga, Emotional Abuse,, Mature Themes,, Physical Abuse,, Suicide,, Violence,, Domestic Abuse, Natsuki Takaya'

In [81]:
data_m_mini.drop("index",axis = 1, inplace= True)

/var/folders/_w/3rp8x21n1fzc11xml_g_c1t80000gn/T/ipykernel_13641/4058840248.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data_m_mini.drop("index",axis = 1, inplace= True)


In [82]:
data_m_mini["Tags"][0]

'Action, Adventure, Fantasy, Shounen, Demons, Historical, Martial Arts, Orphans, Siblings, Swordplay, Based on a Manga, Explicit Violence, Koyoharu Gotouge'

In [83]:
data_m["Tags"][0]

'Action, Adventure, Fantasy, Shounen, Demons, Historical, Martial Arts, Orphans, Siblings, Swordplay, Based on a Manga, Explicit Violence'

In [84]:
data_m

Rank                                               Name   Type  \
0          1  Demon Slayer: Kimetsu no Yaiba - Entertainment...     TV   
1          2                     Fruits Basket the Final Season     TV   
2          3                                    Mo Dao Zu Shi 3    Web   
3          4                   Fullmetal Alchemist: Brotherhood     TV   
4          5                Attack on Titan 3rd Season: Part II     TV   
...      ...                                                ...    ...   
18490  18491  Qin Shi Mingyue: Canghai Hengliu Xiaomeng Spec...    Web   
18491  18492                      Yi Tang Juchang: Sanguo Yanyi     TV   
18492  18493          Fenghuang Ji Xiang Yu Qingming Shanghe Tu     TV   
18493  18494                      Chengshi Jiyi Wo Men de Jieri     TV   
18494  18495  Heisei Inu Monogatari Bow: Genshi Inu Monogata...  Movie   

                                                    Tags  
0      Action, Adventure, Fantasy, Shounen, Demons, H...  
1      Drama, Fantasy, Romance, Shoujo, Animal Transf...  
2      Fantasy, Ancient China, Chinese Animation, Cul...  
3      Action, Adventure, Drama, Fantasy, Mystery, Sh...  
4      Action, Fantasy, Horror, Shounen, Dark Fantasy...  
...                                                  ...  
18490  Action, Ancient China, Chinese Animation, Hist...  
18491                                  Chinese Animation  
18492  Chinese Animation, Family Friendly, Short Epis...  
18493  Chinese Animation, Family Friendly, Short Epis...  
18494                        Comedy, Slice of Life, Dogs  

[13356 rows x 4 columns]

In [85]:
data_m_mini

Rank                                               Name  \
0         1  Demon Slayer: Kimetsu no Yaiba - Entertainment...   
1         2                     Fruits Basket the Final Season   
2         3                                    Mo Dao Zu Shi 3   
3         4                   Fullmetal Alchemist: Brotherhood   
4         5                Attack on Titan 3rd Season: Part II   
...     ...                                                ...   
3194  18444                                      The Imaginary   
3195  18482                                    Goblin Slayer 2   
3196  18484                        Peleliu: Rakuen no Guernica   
3197  18488                                        Make My Day   
3198  18489        Shachiku-san wa Youjo Yuurei ni Iyasaretai.   

                                                   Tags  
0     Action, Adventure, Fantasy, Shounen, Demons, H...  
1     Drama, Fantasy, Romance, Shoujo, Animal Transf...  
2     Fantasy, Ancient China, Chinese Animation, Cul...  
3     Action, Adventure, Drama, Fantasy, Mystery, Sh...  
4     Action, Fantasy, Horror, Shounen, Dark Fantasy...  
...                                                 ...  
3194                    Based on a Novel, A. F. Harrold  
3195  Action, Adventure, Fantasy, Seinen, Dark Fanta...  
3196  Action, Shounen, Island, War, World War 2, Bas...  
3197    Horror, Monsters, Original Work, Yasuo Ootagaki  
3198  Comedy, Slice of Life, Ghosts, Iyashikei, Non-...  

[3199 rows x 3 columns]

In [86]:
data_m.set_index(['Rank','Name'], inplace=True)
data_m.update(data_m_mini.set_index(['Rank','Name']))
data_m.reset_index(inplace=True)
data_m

Rank                                               Name   Type  \
0          1  Demon Slayer: Kimetsu no Yaiba - Entertainment...     TV   
1          2                     Fruits Basket the Final Season     TV   
2          3                                    Mo Dao Zu Shi 3    Web   
3          4                   Fullmetal Alchemist: Brotherhood     TV   
4          5                Attack on Titan 3rd Season: Part II     TV   
...      ...                                                ...    ...   
13351  18491  Qin Shi Mingyue: Canghai Hengliu Xiaomeng Spec...    Web   
13352  18492                      Yi Tang Juchang: Sanguo Yanyi     TV   
13353  18493          Fenghuang Ji Xiang Yu Qingming Shanghe Tu     TV   
13354  18494                      Chengshi Jiyi Wo Men de Jieri     TV   
13355  18495  Heisei Inu Monogatari Bow: Genshi Inu Monogata...  Movie   

                                                    Tags  
0      Action, Adventure, Fantasy, Shounen, Demons, H...  
1      Drama, Fantasy, Romance, Shoujo, Animal Transf...  
2      Fantasy, Ancient China, Chinese Animation, Cul...  
3      Action, Adventure, Drama, Fantasy, Mystery, Sh...  
4      Action, Fantasy, Horror, Shounen, Dark Fantasy...  
...                                                  ...  
13351  Action, Ancient China, Chinese Animation, Hist...  
13352                                  Chinese Animation  
13353  Chinese Animation, Family Friendly, Short Epis...  
13354  Chinese Animation, Family Friendly, Short Epis...  
13355                        Comedy, Slice of Life, Dogs  

[13356 rows x 4 columns]

In [87]:
data_m["Tags"][0]

'Action, Adventure, Fantasy, Shounen, Demons, Historical, Martial Arts, Orphans, Siblings, Swordplay, Based on a Manga, Explicit Violence, Koyoharu Gotouge'

In [88]:
for i in range(0,len(data_m["Tags"])):
    data_m["Tags"][i] = data_m["Tags"][i].replace(" ","")

/var/folders/_w/3rp8x21n1fzc11xml_g_c1t80000gn/T/ipykernel_13641/2813507748.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data_m["Tags"][i] = data_m["Tags"][i].replace(" ","")


In [89]:
# Let us replace , with whitespace
data_m["Tags"].iloc[0]

'Action,Adventure,Fantasy,Shounen,Demons,Historical,MartialArts,Orphans,Siblings,Swordplay,BasedonaManga,ExplicitViolence,KoyoharuGotouge'

In [90]:
for i in range(0,len(data_m["Tags"])):
    data_m["Tags"][i] = data_m["Tags"][i].replace(","," ")

/var/folders/_w/3rp8x21n1fzc11xml_g_c1t80000gn/T/ipykernel_13641/2589152576.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data_m["Tags"][i] = data_m["Tags"][i].replace(","," ")


In [91]:
data_m

Rank                                               Name   Type  \
0          1  Demon Slayer: Kimetsu no Yaiba - Entertainment...     TV   
1          2                     Fruits Basket the Final Season     TV   
2          3                                    Mo Dao Zu Shi 3    Web   
3          4                   Fullmetal Alchemist: Brotherhood     TV   
4          5                Attack on Titan 3rd Season: Part II     TV   
...      ...                                                ...    ...   
13351  18491  Qin Shi Mingyue: Canghai Hengliu Xiaomeng Spec...    Web   
13352  18492                      Yi Tang Juchang: Sanguo Yanyi     TV   
13353  18493          Fenghuang Ji Xiang Yu Qingming Shanghe Tu     TV   
13354  18494                      Chengshi Jiyi Wo Men de Jieri     TV   
13355  18495  Heisei Inu Monogatari Bow: Genshi Inu Monogata...  Movie   

                                                    Tags  
0      Action Adventure Fantasy Shounen Demons Histor...  
1      Drama Fantasy Romance Shoujo AnimalTransformat...  
2      Fantasy AncientChina ChineseAnimation Cultivat...  
3      Action Adventure Drama Fantasy Mystery Shounen...  
4      Action Fantasy Horror Shounen DarkFantasy Isol...  
...                                                  ...  
13351  Action AncientChina ChineseAnimation Historica...  
13352                                   ChineseAnimation  
13353      ChineseAnimation FamilyFriendly ShortEpisodes  
13354      ChineseAnimation FamilyFriendly ShortEpisodes  
13355                            Comedy SliceofLife Dogs  

[13356 rows x 4 columns]

In [92]:
data_m["Tags"].iloc[0]

'Action Adventure Fantasy Shounen Demons Historical MartialArts Orphans Siblings Swordplay BasedonaManga ExplicitViolence KoyoharuGotouge'

In [93]:
data_m["Tags"] = data_m["Tags"].apply(lambda x: x.lower())

In [94]:
# Next let us make a recommendation model

In [95]:
data_m

Rank                                               Name   Type  \
0          1  Demon Slayer: Kimetsu no Yaiba - Entertainment...     TV   
1          2                     Fruits Basket the Final Season     TV   
2          3                                    Mo Dao Zu Shi 3    Web   
3          4                   Fullmetal Alchemist: Brotherhood     TV   
4          5                Attack on Titan 3rd Season: Part II     TV   
...      ...                                                ...    ...   
13351  18491  Qin Shi Mingyue: Canghai Hengliu Xiaomeng Spec...    Web   
13352  18492                      Yi Tang Juchang: Sanguo Yanyi     TV   
13353  18493          Fenghuang Ji Xiang Yu Qingming Shanghe Tu     TV   
13354  18494                      Chengshi Jiyi Wo Men de Jieri     TV   
13355  18495  Heisei Inu Monogatari Bow: Genshi Inu Monogata...  Movie   

                                                    Tags  
0      action adventure fantasy shounen demons histor...  
1      drama fantasy romance shoujo animaltransformat...  
2      fantasy ancientchina chineseanimation cultivat...  
3      action adventure drama fantasy mystery shounen...  
4      action fantasy horror shounen darkfantasy isol...  
...                                                  ...  
13351  action ancientchina chineseanimation historica...  
13352                                   chineseanimation  
13353      chineseanimation familyfriendly shortepisodes  
13354      chineseanimation familyfriendly shortepisodes  
13355                            comedy sliceoflife dogs  

[13356 rows x 4 columns]

In [96]:
data_m.to_csv("ARS_clean", encoding='utf-8', index=False)